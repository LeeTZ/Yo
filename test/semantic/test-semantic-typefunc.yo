type A:
    a: A
    b: A


    type B:
        a: A.B
        b: A

        func eval(self: A.B, c: Int) -> A.B:
            return self

        func eval(self: A.B) -> A.B:
            return self