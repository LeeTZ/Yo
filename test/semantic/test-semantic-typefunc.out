type A
a$A $
b$A $
type B
a$A_B $
b$A $
func eval (self$A_B new$, c$Int new$)
return self$A_B $$A_B $
func eval (self$A_B new$)
return self$A_B $$A_B $